docs/notebooks/multi_objective_ehvi.pct.py

```diff
@@ -0,0 +1,201 @@
+# We consider the VLMOP2 function --- a synthetic benchmark problem with two objectives. We start by defining the problem parameters.
+
+
+# %%
+vlmop2 = VLMOP2().objective()
+observer = trieste.utils.objectives.mk_observer(vlmop2, OBJECTIVE)
+
+# %%
+mins = [-2, -2]
+maxs = [2, 2]
+search_space = Box(mins, maxs)
+num_objective = 2
+
+# %% [markdown]
+# Let's randomly sample some initial data from the observer ...
+
+# %%
+num_initial_points = 10
+initial_query_points = search_space.sample(num_initial_points)
+initial_data = observer(initial_query_points)
+
+# %% [markdown]
+# ... and visualise the data across the design space.
+
+# %%
+_, ax = plot_function_2d(
+    vlmop2,
+    mins,
+    maxs,
+    grid_density=100,
+    contour=True,
+    title=["Obj 1", "Obj 2"],
+    figsize=(12, 6),
+    colorbar=True,
+    xlabel="$X_1$",
+    ylabel="$X_2$",
+)
+plot_bo_points(initial_query_points, ax=ax[0, 0], num_init=num_initial_points)
+plot_bo_points(initial_query_points, ax=ax[0, 1], num_init=num_initial_points)
+plt.show()
+
+# %% [markdown]
+# ... and in the objective space
+
+# %%
+plot_mobo_points_in_obj_space(initial_data[OBJECTIVE].observations)
+plt.show()
+
+
+# %% [markdown]
+# ## Modelling the two functions
+#
+# In this example we model the two objective functions individually with their own Gaussian process models. For problems where the objective functions are similar it may make sense to build a joint model rather than a model stack.
+
+
+# %%
+def build_stacked_independent_objectives_model(data: Dataset, num_output) -> ModelStack:
+        gprs =[]
+        for idx in range(num_output):
+            single_obj_data = Dataset(data.query_points, tf.gather(data.observations, [idx], axis=1))
+            variance = tf.math.reduce_variance(single_obj_data.observations)
+            kernel = gpflow.kernels.Matern52(variance)
+            gpr = gpflow.models.GPR((single_obj_data.query_points, single_obj_data.observations), kernel, noise_variance=1e-5)
+            gpflow.utilities.set_trainable(gpr.likelihood, False)
+            gprs.append((create_model({
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+            "minimize_args": {"options": dict(maxiter=100)}}}), 1))
+
+        return ModelStack(*gprs)
+
+
+# %%
+models = {OBJECTIVE: build_stacked_independent_objectives_model(initial_data[OBJECTIVE], num_objective)}
+
+# %% [markdown]
+# ## Define the acquisition function
+# Here we utilize the [EHVI](https://link.springer.com/article/10.1007/s10898-019-00798-7): `ExpectedHypervolumeImprovement` acquisition function:
+
+# %%
+ehvi = ExpectedHypervolumeImprovement()
+rule: EfficientGlobalOptimization[Box] = EfficientGlobalOptimization(builder=ehvi.using(OBJECTIVE))
+
+# %% [markdown]
+# ## Run the optimization loop
+#
+# We can now run the optimization loop
+
+# %%
+num_steps = 20
```

Comment:
I would like our plots to look a bit better, i.e. like those at the bottom of https://botorch.org/tutorials/multi_objective_bo.

I think this would be achieved with a longer num_steps. perhaps steps=50. How long would that take to run?

Reply:
> I would like our plots to look a bit better, i.e. like those at the bottom of https://botorch.org/tutorials/multi_objective_bo.

Indeed, we could have a color bar representing different bo iterations like in the figure maybe I can update this PR to have in plot functionality directly.

> 
> I think this would be achieved with a longer num_steps. perhaps steps=40. How long would that take to run?

On my laptop, 40 `num_iter` takes 1min20s.

Reply:
I propose setting num_steps to 30 and num_initial_points to 20. This gives us a clearer picture of the Pareto front but isnt too expensive.